patches/server/0465-Add-additional-open-container-api-to-HumanEntity.patch

```diff
@@ -3,47 +3,155 @@ From: JRoy <joshroy126@gmail.com>
++        this.getHandle().openMenu(((CraftingTableBlock) Blocks.CRAFTING_TABLE).getMenuProvider(null, this.getHandle().level(), CraftLocation.toBlockPosition(location)), title); // Paper - overridable title
+         if (force) {
+             this.getHandle().containerMenu.checkReachable = false;
+         }
+@@ -366,7 +366,7 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+     }
+ 
+     @Override
+-    public InventoryView openEnchanting(Location location, boolean force) {
++    public InventoryView openEnchanting(Location location, boolean force, net.kyori.adventure.text.Component title) { // Paper - overridable title
+         if (location == null) {
+             location = getLocation();
+         }
+@@ -379,7 +379,7 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+ 
+         // If there isn't an enchant table we can force create one, won't be very useful though.
+         BlockPos pos = CraftLocation.toBlockPosition(location);
+-        this.getHandle().openMenu(((EnchantmentTableBlock) Blocks.ENCHANTING_TABLE).getMenuProvider(null, this.getHandle().level(), pos));
++        this.getHandle().openMenu(((EnchantmentTableBlock) Blocks.ENCHANTING_TABLE).getMenuProvider(null, this.getHandle().level(), pos), title); // Paper - overridable title
+ 
+         if (force) {
+             this.getHandle().containerMenu.checkReachable = false;
+@@ -422,14 +422,14 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+     }
+ 
+     @Override
+-    public InventoryView openMerchant(Villager villager, boolean force) {
++    public InventoryView openMerchant(Villager villager, boolean force, net.kyori.adventure.text.Component title) { // Paper - overridable title
+         Preconditions.checkNotNull(villager, "villager cannot be null");
+ 
+-        return this.openMerchant((Merchant) villager, force);
++        return this.openMerchant((Merchant) villager, force, title); // Paper - overridable title
+     }
+ 
+     @Override
+-    public InventoryView openMerchant(Merchant merchant, boolean force) {
++    public InventoryView openMerchant(Merchant merchant, boolean force, net.kyori.adventure.text.Component title) { // Paper - overridable title
+         Preconditions.checkNotNull(merchant, "merchant cannot be null");
+ 
+         if (!force && merchant.isTrading()) {
+@@ -455,12 +455,76 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+             throw new IllegalArgumentException("Can't open merchant " + merchant.toString());
+         }
+ 
++        // Paper start - overridable title
++        if (title != null) {
++            name = io.papermc.paper.adventure.PaperAdventure.asVanilla(title);
++            level = 0; // Set level to 0 to remove title suffix
++        }
++        // Paper end
```

Comment:
Not a fan of this because it just overwrites what they entity might have the level set to. I didn't know the title came with a suffix. Calling it `name` then is fine since its not the full title. People can manipulate the `Merchant` if they want to change the level. (I see there's no API for that for the custom merchants, just the entities but that can be added)

Reply:
For the CustomMerchant that is irrelevant because they never append a suffix at all